library/core/src/sync/atomic.rs

```diff
@@ -107,6 +107,33 @@
 //! }
 //! ```
 //!
+//! Although the most convenient way to use an atomic type is to wrap it as an `Arc`,
+//! it is possible to share an atomic type between threads without `Arc`.
+//!
+//! ```
+//! use std::sync::atomic::{AtomicUsize, Ordering};
+//! use std::thread;
+//!
+//! fn main() {
+//!     let data: [usize; 20] = b"Fearless Concurrency".map(Into::into);
```

Comment:
This should probably be slightly less "cute" about the signature of mapping bytes into usizes. It's a clever trick but it's utterly mystifying to drop that one into the middle of Rust's primary API documentation. At the very least, a lot of people won't stop and realize `b"Fearless Concurrency"` is effectively a `[u8; 20]` here, and `.map(Into::into)` is quite opaque as to what type it is converting from and what the destination is.